### ISSUE TITLE
debugger not work when attach IIS process (w3wp.exe)

### ISSUE THREAD

**Issue by jrt324** [0xd4d/dnSpy#100]
Attach to process  IIS  process w3wp.exe, set the dll decomplie code breakpints , the debugger never hit breakpoints

**Comment by 0xd4d** [comment]
Are you sure the code executes over the breakpoint without stopping? Switch to IL view and set a BP at offset 0 of the method.

Did you set the breakpoint in the correct file? Eg. there are many mscorlib.dll files on your harddrive, you must set a breakpoint in the one used by the debugged file. Open the modules window and double click the modules in there to get the correct files.

**Comment by 0xd4d** [comment]
And if possible, it's better to start the debugged program from dnSpy instead of attaching to the process. Otherwise you're debugging optimized code and won't see all locals and you'll have other problems due to CLR optimizations.

**Comment by jrt324** [comment]
OK, it worked!

**Comment by skyyearxp** [comment]
Debug->Modules to select dll to debug, then set breakpoint.

**Comment by silviu-sterian** [comment]
Hi, how did you make it work? I have the same issue, when I Attach to w3p.exe the breakpoints are not being hit. Thanks

**Comment by 0xd4d** [comment]
Most likely you're not setting breakpoints in the correct module. Open the modules window and double click the module so it opens in Assembly Explorer, then set your BPs.

It could also be that it's a small method that gets inlined by the jitter. In that case you can click the Info icon in the attach to process dialog box and read the article that gets opened in your browser.

**Comment by silviu-sterian** [comment]
Thank you!  It worked by opening modules (After attaching the process) and selecting the module.  I was able to do that before reading your response (found somewhere else on the internet) but I really appreciate answering the question. I did not know about inlining of the small methods. Now breakpoints are being hit but I cannot see the local variables. I think I read something about modifying Debug attributes on the dll and recompiling as a fix for that.
